src/trace_processor/trace_summary/summary.cc

```diff
@@ -346,8 +347,42 @@ base::Status Summarize(TraceProcessor* processor,
   }
 
   base::FlatHashMap<std::string, Metric> queries_per_metric;
-  if (!computation.v2_metric_ids.empty()) {
-    for (const auto& id : computation.v2_metric_ids) {
+  if (!computation.v2_metric_ids.has_value()) {
+    // If nullopt, compute all metrics.
+    for (const auto& spec : spec_decoders) {
+      for (auto it = spec.metric_spec(); it; ++it) {
+        protos::pbzero::TraceMetricV2Spec::Decoder m(*it);
+        std::string id = m.id().ToStdString();
+        if (id.empty()) {
+          return base::ErrStatus(
+              "Metric with empty id field: this is not allowed");
+        }
+        if (base::CaseInsensitiveEqual(id, "all")) {
+          return base::ErrStatus("Metric with id 'all' is not allowed: ");
```

Comment:
Can you explain what went wrong here and how the user can fix it.

Ideally this would also be factored out into a funcciton so this can grow tith time.